File: backend/QueueLanka.API/Models/ServiceCenter.cs
namespace QueueLanka.API.Models;

public class ServiceCenter
{
    public int CenterId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? Description { get; set; }
    public string Timezone { get; set; } = string.Empty;
    public int Capacity { get; set; }
    public TimeSpan OpeningTime { get; set; }
    public TimeSpan ClosingTime { get; set; }
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

File: frontend/src/types/serviceCenter.ts
// Service Center Types - SCRUM-25

export interface ServiceCenter {
  centerId: number;
  name: string;
  address: string;
  phone?: string;
  email?: string;
  description?: string;
  timezone: string;
  capacity: number;
  averageServiceTimeMinutes: number;
  openingTime: string;
  closingTime: string;
  isAvailable: boolean;
  isActive: boolean;
  createdAt: string;
}

/** Wrapper shape returned by all backend ApiResponse<T> endpoints */
export interface ApiResponse<T> {
  success: boolean;
  data: T;
  message?: string;
  metadata?: {
    totalCount?: number;
    correlationId?: string;
    timestamp?: string;
  };
}

/** Request payload for POST /api/service-centers (admin only) — SCRUM-67 */
export interface CreateServiceCenterRequest {
  name: string;
  address: string;
  phone?: string;
  email?: string;
  description?: string;
  timezone: string;
  capacity: number;
  averageServiceTimeMinutes: number;
  openingTime: string;
  closingTime: string;
  isActive: boolean;
}

export interface ServiceCenterFilters {
  isAvailable?: boolean;
  isActive?: boolean;
  searchQuery?: string;
}

export interface ServiceCenterApiError {
  code: string;
  message: string;
}

export interface ServiceCenterListResponse {
  serviceCenters: ServiceCenter[];
}

export interface CenterLocation {
  centerId: number;
  streetAddress: string;
  city: string;
  district: string;
  province: string;
  postalCode?: string;
  country: string;
  latitude?: number;
  longitude?: number;
  googleMapsUrl?: string;
  landmark?: string;
}

export interface UpsertLocationRequest {
  streetAddress: string;
  city: string;
  district: string;
  province: string;
  postalCode?: string;
  country: string;
  latitude?: number;
  longitude?: number;
  googleMapsUrl?: string;
  landmark?: string;
}